### PR TITLE
Adding temporal-cache library

### PR DESCRIPTION
**Pull request by timkpaine**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by timkpaine** [comment]
- python `@conda-forge/help-python`
not sure about appveyor but otherwise this is good to go

File: recipes/temporal-cache/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 31cccf3830497336884ae1748a60011e723cd6d312789b6c049bec6ddb3e0f89
+
+build:
+  number: 0
+  skip: True  # [py<37]
```
Comment (npavlovikj) [inline recipes/temporal-cache/meta.yaml]: I think this recipe can be `noarch`, so please remove the `skip` line, and add `noarch: python` instead.

File: recipes/temporal-cache/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
```
Comment (npavlovikj) [inline recipes/temporal-cache/meta.yaml]: With `noarch` recipe, you should use `python >=3.7`.

File: recipes/temporal-cache/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - pip
+  run:
+    - python
```
Comment (npavlovikj) [inline recipes/temporal-cache/meta.yaml]: With `noarch` recipe, you should use `python >=3.7`.

File: recipes/temporal-cache/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  imports:
+    - temporalcache
+
+about:
+  home: http://github.com/timkpaine/temporal-cache
+  license: Apache-2.0
+  license_family: Apache
+  license_file: LICENSE
+  summary: 'Time-based cache invalidation'
+
```
Comment (npavlovikj) [inline recipes/temporal-cache/meta.yaml]: Please remove empty line when modifying the recipe.

File: recipes/temporal-cache/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  description: |
+    Time-based cache invalidation
+  dev_url: https://github.com/timkpaine/temporal-cache
+
```
Comment (npavlovikj) [inline recipes/temporal-cache/meta.yaml]: You can also add `doc_url: https://temporal-cache.readthedocs.io` here.

**Comment by timkpaine** [comment]
@conda-forge/help-python I can't rerun appveyor but it looks unrelated

**Comment by timkpaine** [comment]
@conda-forge/help-python bump

**Comment by synapticarbors** [comment]
Can you do me a favor and merge the latest upstream of `staged-recipes/master` into your branch? It will bring in the latest CI settings and will also kick off the CIs again. I'm pretty sure the error was transient, but just to be sure.

**Pull request merged**